src/OpenTelemetry.Api/Logs/Logger.cs

```diff
@@ -63,8 +68,10 @@ public abstract void EmitLog(
         in LogRecordAttributeList attributes);
 
     internal void SetInstrumentationScope(
-        string? version)
+        string? version,
+        IEnumerable<KeyValuePair<string, object?>>? attributes)
     {
         this.Version = version;
+        this.Attributes = attributes;
```

Comment:
If we are going to do this we need to take a copy here so they can't be mutated after being set.

Same idea as here, which also sorts:

https://github.com/dotnet/runtime/blob/fcd9339f16500a4954d84e9d0d93a99e84b08987/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/ActivitySource.cs#L42-L49